solar-system.rb

```diff
@@ -0,0 +1,197 @@
+    @planets.each do |planet| #an instance of Planet class
+         planets_array << "#{n}.    #{planet.planet_name}"
+         #this is using the get method for planet to return the planet name
+         #as defined in the initialize
+         n += 1
+    end
+    #retunrs the planets array
+    return planets_array
+  end
+
+  #adds a planet to the instance variable of planets through user input
+  def add_planet
+     #allows user to input stats about the planet
+     print "What is the name of the new planet?"
+     planet_name = $stdin.gets.chomp.to_s
+     print "What is the diameter of the planet?"
+     planet_diameter = $stdin.gets.chomp.to_s
+     print "What is the mass of the planet relative to earth?"
+     mass_earth = $stdin.gets.chomp.to_i
+     print "How many moons does the planet have?"
+     number_of_moons = $stdin.gets.chomp.to_i
+     print "How many millions of kilometers is the planet from the sun?"
+     distance_from_the_sun = $stdin.gets.chomp.to_i
+     print "How long is it's year, relative to an earth year?"
+     earth_year = $stdin.gets.chomp.to_f
+
+
+     #creates a new instance of the planet and adds it to
+     #instance variable of Solar System class planets as a class of Planet
+     user_input_planet = Planet.new(planet_name,planet_diameter,mass_earth,number_of_moons, distance_from_the_sun, earth_year)
+       @planets.push(user_input_planet)
+
+     # the user can choose what step is next after adding a planet
+     user_choice
+  end
+
+#lists all planets that are available, and let's the user choose a planet
+#to read about
+  def user_choice
+     #for some reason list_planets would only work with a puts command
+     puts list_planets
+    print "This a list of possible planets.\n"
+
+    print
+    #user chooses planet
+    print "Please type the name of the planet you would like to read about.\n"
+    user_planet = $stdin.gets.chomp.strip
+
+    #for each element in @planets instance variable
+    #prints out a string with the pertinent information
+    @planets.each do |planet|
+        if planet.planet_name == user_planet
+          print "The planet named #{planet.planet_name} , is #{planet.planet_diameter} in diameter, has a mass #{planet.mass_earth} times relative to earth, #{planet.number_of_moons} moons, is #{planet.distance_from_the_sun} million kilometers from the sun, and has a year that is #{planet.earth_year} the length of a year on earth.\n"
+        end
+    end
+
+    #after choice has been made, allows user to pick to continue program
+    #add a planet to the solarsyste or exit the program
+    print "Would you:
+    A: Like to learn about a planet
+    B: Create a planet
+    C: Find the distance of two planets from each other
+    D: Exit\n"
+    user_answer = $stdin.gets.chomp.downcase
+
+    #allows user to learn about more planets
+    #the planet name can not start with a for this to work
+    if user_answer.include? "a"
+        user_choice
+    #allows user to add a planet
```

Comment:
`user_choice` is calling itself here! This is an example of a programming technique called recursion, which you'll learn about in CS Fun in a few months.

Recursion is a powerful tool that can elegantly solve many problems. However, in this case I think a `while` loop might be a better choice, if only because it makes it immediately obvious to the reader that this code might execute many times.